### DISCUSSION TITLE
enable semihosting

### DISCUSSION THREAD

**Discussion by W-M-D** [modm-io/modm#722] [Ideas]
Hello i was wondering how i could enable semi hosting with modm ?

**Comment by salkinium** [comment]
With the [`modm:platform:itm` module](https://modm.io/reference/module/modm-platform-itm/), but be aware of its limitations!

**Comment by salkinium** [comment]
There's also [an example here](https://github.com/modm-io/modm/blob/develop/examples/nucleo_f103rb/itm/main.cpp).

**Comment by salkinium** [comment]
Technically this only provides logging, true semihosting (with  forwarding of libc functions for filesystem access etc) isn't implemented.

**Comment by W-M-D** [comment]
Wow thanks for the quick response!

**Comment by W-M-D** [comment]
Hm it seems that the module isn't available for the stm32f071x8t. Is there any way i can enable it ? I know it exists since i have used it with c before.

**Comment by salkinium** [comment]
Oh, then there's no support for that in modm. 😧 The F0 is using a Cortex-M0, which is lacking the ITM for asynchronous logging, so the fallback would be to use semihosting IO for synchronous logging, but that's not implemented.

**Comment by W-M-D** [comment]
ooofff

**Comment by salkinium** [comment]
Booooooo! Using the semihosting IO solution means stopping the MCU during the data transfer (incl. interrupts), and that wasn't acceptable for most of our applications. So I prioritized the SWO async implementation.

**Comment by W-M-D** [comment]
Oh that might be the reason for a lot of my issues...

**Comment by salkinium** [comment]
I'm closing this, since you've found another solution.
